### PR TITLE
Add associations between book, person and rental 

### PR DESCRIPTION
**Pull request by DavidEspinoG**
# What is new? 
- Add classroom Class
- Add book Class
- Add rental Class as a has-many-through table 
- Add associations between these classes 
### Project requirements
- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -11,9 +11,3 @@ def play_hooky
     '¯(ツ)/¯'
   end
 end
-
-student = Student.new(9, 'David', classroom: 3, parent_permission: false)
```
Comment (OyePriscilla) [inline student.rb:15]: - [ ] In accordance to the project requirement, kindly add `a method that serves as the 'belongs-to' side (a student belongs to a classroom).` 👇 as stated below.

Untitled

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (OyePriscilla) [inline book.rb]: - [ ] According to the project requirement, kindly implement a method to create a new rental using the Rentals class.

```` 
def add_rental (person, date)
    Rental.new(date, self, person)
  end
````

File: person.rb [hunk 3]
```diff
@@ -2,14 +2,15 @@
 
 class Person < Nameable
   attr_reader :id
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
```
Comment (OyePriscilla) [inline person.rb:5]: - [ ] According to the project requirement, kindly implement a method to create a new rental using the `Rentals` class.

 ```
def rent_book(book, date)
    Rental.new(date, book, self)
  end
```

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class Rental
+  attr_accessor :date
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
```
Comment (OyePriscilla) [inline rental.rb:7]: - [ ] The rentals suppose to add itself to @book and @person in order to achieve the `belongs-to` association according to the project requirement. ℹ️ Kindly implement it. 👍 For example:
```
    book.rentals << self    
```

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @DavidEspinoG 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Observed Gitflow ✔️ 
Implement classes in Ruby ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by IngDavidTM — APPROVED** [review]
Hi @DavidEspinoG 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**